hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionFileSystem.java

```diff
@@ -648,11 +651,35 @@ public Path commitDaughterRegion(final RegionInfo regionInfo)
       if (!rename(daughterTmpDir, regionDir)) {
         throw new IOException("Unable to rename " + daughterTmpDir + " to " + regionDir);
       }
+      loadRegionFilesIntoStoreTracker(regionDir);
     }
-
     return regionDir;
   }
 
+  private void loadRegionFilesIntoStoreTracker(Path regionDir) throws IOException {
```

Comment:
We need to use list here? I think when generating the store files when splitting or merging, we should know the store file list?

Reply:
Sorry, missed your ping there. Let me take a look!

Reply:
We do have, but this happens at SplitTableRegionProcedure. We can change some method signature to expect the list of split files as parameters. Do you feel this FS list can cause much of a performance impact here?

Reply:
I tried to get the list of splits/merged files form the split/merge procedure, but it doesn't work if we don't have this on top of the changes from HBASE-26187. This is because without HBASE-26187, these list of files would still be under .split or .merges temp dirs. However, we are doing a renaming in the commit method before trying to load into the tracker, so then when trying to create storefileinfo to add to tracker, the path we have for splits/merges is not valid anymore.

Reply:
I think here we need a big refactoring. First we need to create a StoreFileTracker for the parent region(s), or maybe even StoreEngine, and then load the store file list from the StoreFileTracker, and then, we do split/merge, link the store files to the sub region data directory(s), and then, create StoreFileTrakcer or StoreEngine for the sub region(s), insert the store file list to it(them).

Reply:
Even with all that refactoring, I would still have the same problem: When link files are created for split/merge regions, these are created under temp dir (currently, without HBASE-26187). So even with a list of all those references in memory, we cannot simply add this list to the StoreFileTracker, because these links in the list are under a temp directory. We would need to iterate through this list and update the links paths to point to the actual directory (after the region dir has been renamed), at commit time, before we can add the list to tracker.

Reply:
I've rebased HBASE-26067, now it should include HBASE-26187 now~

Please have a try~